### PR TITLE
[MM-27919] Allow i18n and resources to be loaded using an environment variable

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This modifies the test resource loading and the `i18n` files to be loaded from a dynamic location using an environment variable. This mechanism will be used by `mmctl` to start the server reusing the same test helpers that we have for the API to run end to end tests and to be able to use the app layer within them.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-27919

### CODE REVIEW

**Review by jespino — APPROVED** [review]
LGTM

File: testlib/resources.go [hunk 1]
```diff
@@ -36,17 +36,26 @@ type testResourceDetails struct {
...
+	paths := []string{
+		".",
+		"..",
+		"../..",
+		"../../..",
+		"../../../..",
+	}
+
+	// this enables the server to be used in tests from a different repository
+	if mmPath := os.Getenv("MM_SRC_PATH"); mmPath != "" {
```
Comment (jespino) [inline testlib/resources.go:50]: Maybe "MM_SERVER_PATH"? or something less ambiguous?

Comment (mgdelacroix) [reply testlib/resources.go]: makes sense, modified!

**Pull request merged**